README.md

```diff
@@ -3,107 +3,178 @@
 
 ### Defining php-fpm pools
 
-If different php-fpm pools are required, you can use `php::fpm::pool`
-resources. Use the parameter `$pools` from the class `php::fpm` to define
-`php::fpm::pool` resources as a hash from hiera.
+If different php-fpm pools are required, you can use `php::fpm::pool` defined resource type. A single pool called `www` will be configured by default. Specify additional pools like so:
 
-If not defined in your hiera data, a single pool called `www` will be
-configured by default:
-
-```yaml
-php::fpm::pools:
-  www:
-    listen: 127.0.0.1:9000
+```puppet
+  php::fpm::pool { 'www2':
+    listen => '127.0.1.1:9000',
+  }
 ```
 
 For an overview of all possible parameters for `php::fpm::pool` resources
 please see [its documention](http://php.puppet.mayflower.de/php/fpm/pool.html).
 
+### A complete "Roles & Profiles" style example with Hiera lookups
+This is all of the above examples put together. It is typical of what you would see with "Roles & Profiles" style of Puppet design, which is what Puppet Labs currently regards as their [unofficial] "best practice." The data values are stored in YAML format and passed into puppet DSL with `hiera` lookups.
+
+##### Hiera
+```yaml
+---
+php::ensure:       'latest'
+php::manage_repos: 'true'
+php::fpm:          'true'
+php::dev:          'true'
+php::composer:     'true'
+php::pear:         'true'
+php::phpunit:      'false'
+php::settings:
+ 'PHP/max_execution_time':  '90'
+ 'PHP/max_input_time':      '300'
+ 'PHP/memory_limit':        '64M'
+ 'PHP/post_max_size':       '32M'
+ 'PHP/upload_max_filesize': '32M'
+ 'Date/date.timezone':      'Europe/Berlin'
+php::extensions:
+ bcmath: {}
+ xmlrpc: {}
+ imagick:
+  provider: 'pecl'
+ memcached:
+  provider: 'pecl'
+  header_packages:
+   - 'libmemcached-devel'
+ apc:
+  provider: 'pecl'
+  settings:
+   'apc/stat':       '2'
+   'apc/stat_ctime': '2'
+php::fpm::pool:
+ 'www2':
+  listen: '127.0.1.1:9000'
+```
+
+##### Puppet Profile
+```puppet
+class profile::php {
+  
+  # Hiera lookups
+  $php_ensure       = hiera('php::ensure')
+  $php_manage_repos = hiera('php::manage_repos')
+  $php_fpm          = hiera('php::fpm')
+  $php_dev          = hiera('php::dev')
+  $php_composer     = hiera('php::composer')
+  $php_pear         = hiera('php::pear')
+  $php_phpunit      = hiera('php::phpunit')
+  $php_settings     = hiera('php::settings')
+  $php_extensions   = hiera('php::extensions')
+  $php_fpm_pool     = hiera('php::fpm::pool')
+  
+
+  # PHP Class
+  class { '::php':
+    ensure       => $php_ensure,
+    manage_repos => $php_manage_repos,
+    fpm          => $php_fpm,
+    dev          => $php_dev,
+    composer     => $php_composer,
+    pear         => $php_pear,
+    phpunit      => $php_phpunit,
+    extensions   => $php_extensions,
+    settings     => $php_settings,
+  }
+  
+  # Create php-fpm pools
+  create_resources(php::fpm::pool, $php_fpm_pool)
+
+}
+```
+
```

Comment:
`class profile::php` is not an "extra profile wrapping class."

Yes I agree that it is totally unnecessary for the operation of your module. However, may I please direct your attention to the heading of this section. As you can see I was attempting to show an example of how to use this the R&P style of puppet coding which is growing in popularity. Myself and others have been mentioning it in various issues/PRs in this repo.

Ideally of course this documentation should be "puppet coding style" agnostic. My reasoning for adding this section is this module's hiera-centric history. I thought it would be a good idea to add this example to help folks that come here looking for how to properly use it with hiera.

Now that I said my piece, if you still want me to remove this section just say the word